draft-ietf-quic-http.md

```diff
@@ -912,10 +912,20 @@ Endpoints that set low values for the QUIC transport parameters
 chance that the remote peer reaches the limit early and becomes blocked. In
 particular, the value chosen for `initial_max_uni_streams` should consider that
 remote peers may wish to exercise reserved stream behaviour ({{stream-grease}}).
-To reduce the likelihood of blocking, both clients and servers SHOULD send a
-value of three or greater for the QUIC transport parameter
-`initial_max_uni_streams`, and a value of 1,024 or greater for the QUIC
-transport parameter `initial_max_stream_data_uni`.
+To reduce the likelihood of blocking, both clients and servers MUST allow
+the peer to create one unidirectional stream for the HTTP control stream plus
+the number of unidirectional streams required by mandatory extensions (such as
+QPACK), by setting an appropriate value for the QUIC transport parameter
```

Comment:
This comma doesn't flow as-is.  I'd try one of the following:
- remove it ("...MUST allow ... by setting an appropriate value....")
- make it a lead-in to options ("...MUST allow [...], either by setting an appropriate value for [...] or by sending a MAX_STREAMS frame immediately after opening the connection.")
- split into a new sentence ("...MUST allow [...].  This can be done by setting....")

The implicit question in those options is whether omitting the TP and immediately sending a MAX_STREAMS frame is acceptable behavior.

Reply:
I'd say you have to put it in the TP.  What is the possible reason an implementation would have for setting inital_max_uni_streams to less than 3 and making up for it with MAX_STREAMS?  What if the packet containing MAX_STREAMS is lost?

Reply:
I went with option 1. I also agree that the TP is the best way. Also because the paragraph now starts with "To avoid blocking", and sending MAX_STREAMS is not a great way to avoid blocking.